### PR TITLE
[READY] move min, max, minmax, clamp to math functions #4425

### PR DESCRIPTION
**Pull request by fnrizzi**
addresses #4425

### CODE REVIEW

**Comment by fnrizzi** [comment]
@dalg24 is this the right place to move these functions to? I kept everything under experimental as agreed on.

**Comment by dalg24** [comment]
> @dalg24 is `Kokkos_MathematicalFunctions.hpp` the right place to move these functions to? I kept everything under experimental as agreed on.

I'd keep them in their own header like you had them before.

**Comment by fnrizzi** [comment]
@dalg24 ok, thanks, but in their own header inside `core` correct?

**Comment by dalg24** [comment]
> @dalg24 ok, thanks, but in their own header inside `core` correct?

Yes

File: core/src/Kokkos_MinMaxClamp.hpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+#include <Kokkos_Macros.hpp>
+
+namespace Kokkos {
+
+// this fwd declaration is here because paid
+// is needed below and if we don't do this,
+// the compiler does not find it
+template <class T1, class T2>
+struct pair;
```
Comment (dalg24) [inline core/src/Kokkos_MinMaxClamp.hpp:56]: Why a forward declaration rather than including `<Kokkos_Pair.hpp>`?

Comment (fnrizzi) [reply core/src/Kokkos_MinMaxClamp.hpp]: this was from before when i put them inside MathFunctions, I did not want to add another include. 
But I added the include!

File: core/unit_test/CMakeLists.txt [hunk 2]
```diff
@@ -117,6 +117,7 @@ foreach(Tag Threads;Serial;OpenMP;Cuda;HPX;OpenMPTarget;HIP;SYCL)
         FunctorAnalysis
         Init
         LocalDeepCopy
+	MinMaxClamp
```
[the remark this answers is not in the document]
Comment (fnrizzi) [reply core/unit_test/CMakeLists.txt]: done!

File: algorithms/src/Kokkos_StdAlgorithms.hpp [hunk 3]
```diff
@@ -55,9 +55,6 @@
 // distance
 #include <std_algorithms/Kokkos_Distance.hpp>
 
-// clamp, min, max, minmax
-#include <std_algorithms/Kokkos_MinMaxOperations.hpp>
```
[the remark this answers is not in the document]
Comment (fnrizzi) [reply algorithms/src/Kokkos_StdAlgorithms.hpp:59]: no, it is removed

[the remark this answers is not in the document]
Comment (dalg24) [reply algorithms/src/Kokkos_StdAlgorithms.hpp:59]: https://github.com/kokkos/kokkos/blob/59d0f57ea00f5fd7355e55f8c8fc93fee1859db3/algorithms/src/std_algorithms/Kokkos_MinMaxOperations.hpp

[the remark this answers is not in the document]
Comment (fnrizzi) [reply algorithms/src/Kokkos_StdAlgorithms.hpp:59]: ok, done, sorry Damien! I am a bit under the weather today. Will be more careful.

**Review by cz4rs — APPROVED** [review]
Looks good to me!

**Comment by cz4rs** [comment]
I think this should be good to go once #4509 is merged.

**Comment by masterleinad** [comment]
Should the title still contain "[WIP]"?

**Comment by fnrizzi** [comment]
> Should the title still contain "[WIP]"?

@masterleinad I left it there because I saw CI partially failing, so was waiting to see if was something caused by the PR. 
I think I can remove the WIP since it seems it is not that.

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
Retest this please

**Review by crtrott — APPROVED** [review]

**Pull request merged**